### PR TITLE
Fix aus-map TS errors

### PR DESCRIPTION
**Pull request by marialani**
## What are you doing in this PR?
This fixes Typescript and linting errors for the Aus moment map. The map has been tested locally and in code.

[**Trello Card**](https://trello.com/c/uT6ggVQR/213-typescript-error-busting-aus-moment-map)

### CODE REVIEW

**Comment by marialani** [comment]
[Typescript Error Busting - aus moment map](https://trello.com/c/uT6ggVQR/213-typescript-error-busting-aus-moment-map)

File: support-frontend/assets/pages/aus-moment-map/ausMomentMap.tsx [hunk 1]
```diff
@@ -145,7 +160,12 @@ const AusMomentMap = () => {
 					{windowWidthIsLessThan('desktop') && <Blurb />}
 					<Map
 						selectedTerritory={selectedTerritory}
-						setSelectedTerritory={(territory) => {
+						setSelectedTerritory={(
+							territory:
+								| string
+								| ((prevState: string | null) => string | null)
+								| null,
+						) => {
```
Comment (TomPretty) [inline support-frontend/assets/pages/aus-moment-map/ausMomentMap.tsx:168]: I think from the `Map` components perspective it should only need a function that's `(territory: string) => void`

File: support-frontend/assets/pages/aus-moment-map/components/blurb.tsx [hunk 2]
```diff
@@ -8,11 +7,13 @@ const enableSupporterCount = true;
 const useSupportersCount = () => {
 	const [supportersCount, setSupportersCount] = React.useState(0);
 	const supportersCountEndpoint = '/supporters-ticker.json';
+
 	React.useEffect(() => {
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- if check here is supposedly unnecessary
```
Comment (TomPretty) [inline support-frontend/assets/pages/aus-moment-map/components/blurb.tsx:12]: so the issue is the 

```ts
enableSupporterCount = true;
```

which typescript then uses to assign the literal type `true` to `enableSupportersCount`. We can force typescript to just class it as a `boolean` with something like:

```ts
enableSupportersCount = true as boolean;
```

File: support-frontend/assets/pages/aus-moment-map/components/map.tsx [hunk 3]
```diff
@@ -12,29 +12,32 @@ import VictoriaSvg from './territories/victoriaSvg';
 import WesternAustraliaSvg from './territories/westernAustraliaSvg';
 
 type TerritorySvgContainerProps = {
-	onClick: () => void;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any -- the expected type: MouseEventHandler<SVGGElement> is not assignable
+	onClick: any;
```
Comment (TomPretty) [inline support-frontend/assets/pages/aus-moment-map/components/map.tsx:16]: I think we should be able to keep the original type here

File: support-frontend/assets/pages/aus-moment-map/components/map.tsx [hunk 4]
```diff
@@ -12,29 +12,32 @@ import VictoriaSvg from './territories/victoriaSvg';
...
 
-const TerritorySvgContainer = (props: TerritorySvgContainerProps) => {
+function TerritorySvgContainer(props: TerritorySvgContainerProps) {
 	const { windowWidthIsGreaterThan } = useWindowWidth();
 	return (
 		<g
 			className={`territory ${props.isSelected ? 'territory-selected' : ''}`}
-			onClick={windowWidthIsGreaterThan('desktop') ? props.onClick : null}
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- the expected type: MouseEventHandler<SVGGElement> is not assignable
+			onClick={windowWidthIsGreaterThan('desktop') && props.onClick}
```
Comment (TomPretty) [inline support-frontend/assets/pages/aus-moment-map/components/map.tsx:27]: and then do something like:

```ts
			onClick={() => {
				windowWidthIsGreaterThan('desktop') && props.onClick();
			}}
```

**Comment by GHaberis** [comment]
@marialani is it possible to resolve the feedback you've responded to and fix the conflicts, then I think this will be good to ship 🚀

File: support-frontend/assets/pages/aus-moment-map/components/map.tsx [hunk 5]
```diff
@@ -14,27 +14,33 @@ import WesternAustraliaSvg from './territories/westernAustraliaSvg';
...
-	children: React.ReactNode;
+	children: ReactNode;
 };
 
-const TerritorySvgContainer = (props: TerritorySvgContainerProps) => {
+function TerritorySvgContainer(props: TerritorySvgContainerProps) {
 	const { windowWidthIsGreaterThan } = useWindowWidth();
+
+	const handleClick = () => {
+		windowWidthIsGreaterThan('desktop') && props.onClick;
```
Comment (GHaberis) [inline support-frontend/assets/pages/aus-moment-map/components/map.tsx:24]: I think we want to call `props.onClick` here:

```
if (windowWidthIsGreaterThan('desktop') {
 props.onClick()
}
```

**Review by GHaberis — APPROVED** [review]
Looks good 👍

**Pull request merged**